libuavcan/include/uavcan/driver/can.hpp

```diff
@@ -42,15 +47,42 @@ struct UAVCAN_EXPORT CanFrame
         fill(data, data + MaxDataLen, uint8_t(0));
     }
 
-    CanFrame(uint32_t can_id, const uint8_t* can_data, uint8_t data_len) :
+    CanFrame(uint32_t can_id, const uint8_t* can_data, uint8_t data_len, bool canfd_frame = false) :
         id(can_id),
-        dlc((data_len > MaxDataLen) ? MaxDataLen : data_len)
+        canfd(canfd_frame)
     {
         UAVCAN_ASSERT(can_data != UAVCAN_NULLPTR);
         UAVCAN_ASSERT(data_len == dlc);
-        (void)copy(can_data, can_data + dlc, this->data);
+        (void)copy(can_data, can_data + data_len, this->data);
```

Comment:
this logic needs to be moved to the cpp file

Reply:
doing that costs extra flash